onnxruntime/core/providers/cpu/math/element_wise_ops.cc

```diff
@@ -216,118 +216,151 @@ ONNX_CPU_OPERATOR_KERNEL(
+      }};
+
+  UntypedBroadcastTwo(*context, funcs, 1.0);
+  return Status::OK();
 }
 
 template <typename T>
 Status Div<T>::Compute(OpKernelContext* context) const {
-  return BroadcastTwo<T, T>(
-      *context,
-      [](EigenVectorMap<T> output, T input0, ConstEigenVectorMap<T> input1) { output = input0 / input1.array(); },
-      [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, T input1) { output = input0.array() / input1; },
-      [](EigenVectorMap<T> output, ConstEigenVectorMap<T> input0, ConstEigenVectorMap<T> input1) { output = input0.cwiseQuotient(input1); },
-      1.0f);
-}
+  ProcessBroadcastSpanFuncs funcs{
+      [](BroadcastHelper& per_iter_bh) {
+        per_iter_bh.OutputEigen<T>() = per_iter_bh.ScalarInput0<T>() / per_iter_bh.EigenInput1<T>().array();
+      },
+      [](BroadcastHelper& per_iter_bh) {
+        per_iter_bh.OutputEigen<T>() = per_iter_bh.EigenInput0<T>().array() / per_iter_bh.ScalarInput1<T>();
+      },
+      [](BroadcastHelper& per_iter_bh) {
+        per_iter_bh.OutputEigen<T>() = per_iter_bh.EigenInput0<T>().cwiseQuotient(per_iter_bh.EigenInput1<T>());
+      }};
 
+  UntypedBroadcastTwo(*context, funcs, 1.0);
+  return Status::OK();
+}
 
 namespace pow_internal {
 
 template <typename T, typename E>
-void PowImpl(OpKernelContext* context, const Tensor& X, const Tensor& Y) {
-  TBroadcaster<T, E> bc{X, Y};
-  Tensor* const output_tensor = context->Output(0, bc.GetOutputShape());
-  TBroadcastOutput<T> output{bc.GetSpanSize(), *output_tensor};
-
-  // Scalar base
-  auto input0scalar = [](gsl::span<T> output, T X, gsl::span<const E> Y) {
-    std::transform(Y.cbegin(), Y.cend(), output.begin(),
-                   [X](E y) {
-                     return static_cast<T>(std::pow(X, y));
-                   });
-  };
-
-  // Scalar exponent switch to possibly available optimizations
-  std::function<void(gsl::span<T>, gsl::span<const T> X, E Y)> input1scalar =
-      [](gsl::span<T> output, gsl::span<const T> X, E Y) {
-        std::transform(X.cbegin(), X.cend(), output.begin(),
-                       [Y](T x) {
-                         return static_cast<T>(std::pow(x, Y));
-                       });
-      };
+void PowImpl(OpKernelContext& context) {
+  ProcessBroadcastSpanFuncs funcs{
+      [](BroadcastHelper& per_iter_bh) {
+        const T X = per_iter_bh.ScalarInput0<T>();
+        auto Y = per_iter_bh.SpanInput1<E>();
+        auto output = per_iter_bh.OutputSpan<T>();
 
-  if (Y.Shape().Size() == 1) {
-    auto exp = *Y.template Data<E>();
-    if (exp == E{2}) {
-      input1scalar = [](gsl::span<T> output, gsl::span<const T> X, E) {
-        std::transform(X.cbegin(), X.cend(), output.begin(),
-                       [](T x) {
-                         return static_cast<T>(x * x);
+        std::transform(Y.cbegin(), Y.cend(), output.begin(),
+                       [X](E y) {
+                         return static_cast<T>(std::pow(X, y));
```

Comment:
These are also lambdas. Would rewriting them as a loop help?

Reply:
Lambdas aren't necessarily problematic. In this case std::transform is templatized so the lambda is not being pushed into a std::function and there's no unnecessary code covered by template. Based on that I wouldn't expect rewriting as a loop to provide much if any benefit.